lib/exec-sh.js

```diff
@@ -82,4 +82,12 @@ function execSh (command, options, callback) {
   return child
 }
 
+execSh.promise = function (command, options) {
+  return new Promise(function (resolve, reject) {
+    execSh(command, options, function (err, stdout, stderr) {
+      (err || stderr) ? reject(err || stderr) : resolve(stdout);
```

Comment:
`(err || stderr)` - `stderr` might be non empty even for successfully executed commands
`reject(err || stderr)` - error object containing exit status might be lost here

Reply:
You're right, I think i should change code to `reject` only if there `err`, otherwise it should always resolve with object like `{ stderr, stdout }`?

Reply:
To stay consistent with current callback interface both `stderr` and `stdout` must be always exposed.

Perhaps storing `stderr` and `stdout` on `err` object as properties would make sense. 

Something like:
```
if (err) {
  err.stdout = stdout
  err.stderr = stderr
  reject(err)
}

resolve({stderr, stdout})
```